packages/node_modules/pouchdb-adapter-indexeddb/src/setup.js

```diff
@@ -115,6 +115,11 @@ function openDatabase(openDatabases, api, opts, resolve, reject) {
       // migrate between the two. In the future, dependent on performance tests,
       // we might silently migrate
       throw new Error('Incorrect adapter: you should specify the "idb" adapter to open this DB');
+    } else if (e.oldVersion === 0 && e.newVersion < versionMultiplier) {
+      // Firefox still creates the database with version=1 even if we throw,
+      // so we need to be sure to destroy the empty database before throwing
+      indexedDB.deleteDatabase(opts.name);
+      throw new Error('Database was deleted while open');
```

Comment:
if you don't mind, can i ask how do you catch this error?
the context is that i'm trying to make the frontend code choose between the idb or indexeddb adapter, in order to migrate users from the old to the new

i was hoping to do something like this:
```javascript
// example that did not work like i hoped
async function getDB () {
  var db = new PouchDB('myDb', { adapter: 'indexeddb' })
  try {
      await db.info()
  } catch (e) {
     if (e.message.startsWith('Incorrect adapter')) {
        return new PouchDB('myDb', { adapter: 'idb' })
     }
  }
}

but because the error is thrown, and not used to reject the promise, it seems like i'm not able to catch it, were you able to do this somehow?

Reply:
Hi , @karlwestin , sorry for the delay in replying
We don't catch specifically this error, but AFAIR, `onupgradeneeded` exceptions passed through some of these asynchronous calls (I think the first time we tried to access it, but not sure), so we could catch it more or less like you did inside an `async` function and it did work:
https://github.com/raiden-network/light-client/blob/master/raiden-ts/src/db/utils.ts#L44-L69